### PR TITLE
pybullet ROM spherical constraint learning

### LINKED ISSUE #43: pybullet unit test is nondeterministic, possibly due to motion planning timeouts

### PR DESCRIPTION
**Pull request by tomsilver**
plus:
1. make all of the motion planning stuff and all of the pybullet tests deterministic, hopefully... (closes #43 ) [note @RealZiangLiu you'll need to re-install `pybullet_helpers` because some of the fixes were in that package]
2. implement reset() in pybullet env
3. move tray out of the way for now so that we don't have to worry about collisions in book handover task
4. add collision checking constraint for possible collisions with the human / wheelchair (saw in seed 5)
5. change spherical ROM sampling to be _within_ sphere instead of _on_ sphere
6. some other small stuff
pybullet_spherical_csp

### CODE REVIEW

File: experiments/conf/config.yaml [hunk 1]
```diff
@@ -8,10 +8,10 @@ hydra:
 video_dir: ${hydra:runtime.output_dir}/videos
 results_file: ${hydra:runtime.output_dir}/results.csv
 seed: 0
-num_episodes: 1000
-max_episode_length: 100
```
Comment (RealZiangLiu) [inline experiments/conf/config.yaml:12]: Since max_episode_length should be environment specific, perhaps it should go under config for each environment?

File: src/multitask_personalization/rom/models.py [hunk 2]
```diff
@@ -237,22 +243,24 @@ def set_trainable_parameters(self, params: Any) -> None:
         self._radius = params
 
     def check_position_reachable(
-        self, position: NDArray, padding: float = 1e-6
+        self, position: NDArray, padding: float = 1e-4
     ) -> bool:
         distance = float(np.linalg.norm(position - self._sphere_center))
-        return distance < self._radius + padding
+        reachable = distance < self._radius + padding
+        return reachable
```
Comment (RealZiangLiu) [inline src/multitask_personalization/rom/models.py:250]: Curious why this change was made.... mainly interpretability?

File: src/multitask_personalization/methods/csp_approach.py [hunk 3]
```diff
@@ -39,9 +46,19 @@ def reset(
             # We will refactor this in a future PR.
             if isinstance(obs, TinyState):
                 self._csp_generator = TinyCSPGenerator(self._seed)
+            elif isinstance(obs, PyBulletState):
+                task_spec = info["task_spec"]
+                assert isinstance(task_spec, PyBulletTaskSpec)
+                sim = PyBulletEnv(task_spec, seed=self._seed, use_gui=False)
+                rom_model = SphericalROMModel(task_spec.human_spec, self._seed)
```
Comment (RealZiangLiu) [inline src/multitask_personalization/methods/csp_approach.py:53]: seed=self._seed for kwarg for consistency

File: src/multitask_personalization/rom/models.py (hunk 2 above)
Comment (tomsilver) [reply src/multitask_personalization/rom/models.py:250]: oops this is just left over from me printing out the value of `reachable` for debugging. I guess I'll leave it in case we want to do that again int he future

File: experiments/conf/config.yaml (hunk 1 above)
Comment (tomsilver) [reply experiments/conf/config.yaml:12]: good call

Comment (tomsilver) [reply experiments/conf/config.yaml:12]: actually it looks like if I put these values in the configs for the environments, there's an error because the values get passed into the environments when we do `hydra.instantiate`. instead, let me figure out how to separate out different top-level configs for different experiments and put the values in there...

Comment (tomsilver) [reply experiments/conf/config.yaml]: okay I followed the example here: https://hydra.cc/docs/patterns/configuring_experiments/
